src/wallet/wallet.cpp

```diff
@@ -1446,6 +1524,38 @@ CAmount CWallet::GetBalance() const
     return nTotal;
 }
 
+CAmount CWallet::GetColdStakingBalance() const
+{
+    CAmount nTotal = 0;
+    {
+        LOCK2(cs_main, cs_wallet);
+        for (std::map<uint256, CWalletTx>::const_iterator it = mapWallet.begin(); it != mapWallet.end(); ++it) {
+            const CWalletTx* pcoin = &(*it).second;
+
+            if (pcoin->IsTrusted())
+                nTotal += pcoin->GetColdStakingCredit();
```

Comment:
speed up this validating the tx type, would add a `pcoin->IsColdStaking(true)` (the bool param tells to the method that the cold staking credit must be mine and not that I have such tx for a MN reward).

Reply:
Sure.
https://github.com/PIVX-Project/PIVX/pull/955/commits/9145fb558998a937538d62056ac6e6b6ea18e314 adds a `HasP2CSOutputs()` check to that.
I don't think that `IsMine` should be checked beforehand though, as that is done inside `GetCredit`.